src/analyses/variable-sensitivity/abstract_environment.cpp

```diff
@@ -196,49 +225,51 @@ bool abstract_environmentt::assume(const exprt &expr, const namespacet &ns)
   // We should only attempt to assume Boolean things
   // This should be enforced by the well-structured-ness of the
   // goto-program and the way assume is used.
-
   PRECONDITION(expr.type().id() == ID_bool);
 
-  // Evaluate the expression
-  abstract_object_pointert res = eval(expr, ns);
-
-  exprt possibly_constant = res->to_constant();
+  auto simplified = simplify_expr(expr, ns);
```

Comment:
`ai_simplify` isn't available here because that's a member function on `variable_sensitivity_domain`. Luckily, we call `assume` from `variable_sensitivity_domain`, so we can just call `assume` first :slightly_smiling_face: So I've done that.

Reply:
Great!